examples/webgl_loader_pcd.html

```diff
@@ -61,19 +67,126 @@
+
+								colors[ i * 3 + 0 ] = c.r;
+								colors[ i * 3 + 1 ] = c.g;
+								colors[ i * 3 + 2 ] = c.b;
+
+							}
+
+							points.geometry.setAttribute( 'color', new THREE.Float32BufferAttribute( colors, 3 ) );
+							if ( points.material && 'vertexColors' in points.material ) points.material.vertexColors = true;
+
+						}
+
+						// For car_scene (Z-up data), use Z-up camera and fit view
+						if ( zUpFiles.includes( file ) ) {
+
+							// Use Z-up for camera and controls
+							camera.up.set( 0, 0, 1 );
+
+							// Recreate OrbitControls to avoid roll from prior Y-up init
+							if ( controls ) controls.dispose();
+							controls = new OrbitControls( camera, renderer.domElement );
+							controls.addEventListener( 'change', render );
+							controls.minDistance = 0.5;
+							controls.maxDistance = 1000;
+							controls.screenSpacePanning = false; // Use world-up panning
+
+							// Compute bounding sphere
+							if ( ! points.geometry.boundingSphere ) points.geometry.computeBoundingSphere();
+							const sphere = points.geometry.boundingSphere;
+							const center = sphere.center.clone();
+							const radius = sphere.radius || 1;
+
+							// Compute distance by FOV to fully frame (fit view)
+							const fov = camera.fov * Math.PI / 180;
+							const distance = radius / Math.sin( fov / 2 );
+
+							// Set near/far to cover the whole cloud
+							camera.near = Math.max( 0.001, distance * 0.01 );
+							camera.far = distance * 10 + radius;
+							camera.updateProjectionMatrix();
+
+							// Set controls target to cloud center
+							controls.target.copy( center );
+
+							// Place camera in an oblique top view for Z-up
+							const viewDir = new THREE.Vector3( 0, - 1, 0.5 ).normalize();
+							camera.position.copy( center ).add( viewDir.multiplyScalar( distance ) );
+							camera.lookAt( center );
+							// Clamp polar angle to avoid flip; update matrices
+							controls.minPolarAngle = 0.01;
+							controls.maxPolarAngle = Math.PI - 0.01;
+							controls.update();
+							camera.updateProjectionMatrix();
+			
+						} else {
+
+							// Non car_scene: keep legacy rotateX(Math.PI) for display
+							points.geometry.rotateX( Math.PI );
+
+							// Restore default camera (Y-up) and basic params
+							camera.up.set( 0, 1, 0 );
+							camera.near = 0.01;
+							camera.far = 40;
+							camera.updateProjectionMatrix();
+
+							// Recreate OrbitControls for default Y-up semantics
+							if ( controls ) controls.dispose();
+							controls = new OrbitControls( camera, renderer.domElement );
+							controls.addEventListener( 'change', render );
+							controls.minDistance = 0.5;
+							controls.maxDistance = 10;
+							controls.screenSpacePanning = true;
+
+							// Restore initial position and target
+							controls.target.set( 0, 0, 0 );
+							camera.position.set( 0, 0, 1 );
+							camera.lookAt( 0, 0, 0 );
+							controls.update();
+							camera.updateProjectionMatrix();
+
+						}
```

Comment:
Do you mind removing these changes from the demo and just add `lidar_scene.pcd` as a new asset?